### PR TITLE
Migrate DHCPv4 tests to new net_pkt API

### PR DESCRIPTION
**Pull request by rveerama1**

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: tests/net/dhcpv4/src/main.c [hunk 1]
```diff
@@ -203,128 +203,40 @@ static void net_dhcpv4_iface_init(struct net_if *iface)
...
-	/* Place the IPv4 header */
-	set_ipv4_header(pkt);
-
-	/* Place the UDP header */
-	set_udp_header(pkt);
 
-	net_buf_add(frag, NET_IPV4UDPH_LEN);
-	offset = NET_IPV4UDPH_LEN;
+	pkt = net_pkt_alloc_with_buffer(iface,
+					sizeof(struct net_ipv4_hdr) +
```
Comment (tbursztyka) [inline tests/net/dhcpv4/src/main.c]: When given family is not AF_UNSPEC, the allocator will take care of counting the header size, same for protocol if it's not 0. So you can remove the ipv4 and udp hdr size

Apply that where relevant

File: tests/net/dhcpv4/src/main.c [hunk 2]
```diff
@@ -401,59 +287,53 @@ struct net_pkt *prepare_dhcp_ack(struct net_if *iface, u32_t xid)
 
 static int parse_dhcp_message(struct net_pkt *pkt, struct dhcp_msg *msg)
 {
-	struct net_buf *frag = pkt->frags;
-	u8_t type;
-	u16_t offset;
+	net_pkt_cursor_init(pkt);
```
Comment (tbursztyka) [inline tests/net/dhcpv4/src/main.c]: I guess this is superfluous, net_send_data() and, if required, l2 will call that already.

**Review by tbursztyka — APPROVED** [review]
Tiny detail.

Rest lgtm. I am approving already since I am not going to be back soon.

File: tests/net/dhcpv4/src/main.c (hunk 1 above)
Comment (rveerama1) [reply tests/net/dhcpv4/src/main.c]: PR merged already, I will do that in separate PR.

Comment (rveerama1) [reply tests/net/dhcpv4/src/main.c]: https://github.com/zephyrproject-rtos/zephyr/pull/13946 fixes issues you mentioned.

**Pull request merged**